_posts/jasonwang/2016-04-29-jasonwang-finalproject.md

```diff
@@ -0,0 +1,51 @@
+
+Between this project and the last, I personally believe that my debugging skill had increased dramatically. While in the last project I had to keep playing around with values (and sometimes breaking it more) to try and isolate why certain functions weren’t working as intended, zeroing in on problems in this project was much easier. This may be helped by the fact that you can track the data as it flows through, but I also think that my deeper understanding of Python since the last project allowed me to stop and think, “Why?”. And from “why”, we can narrow down the possible areas of interest.
+
+Honestly, between the structure of this project and the last project, I thoroughly appreciated the open-ended, milestone-focused aspect of keeping on track. The milestones really did help me quickly narrow down what topics I wanted to than if it was completely free, and the updates pushed me to make progress between each class. While I’m not saying that I would be planning on finishing the whole project 30 minutes before, the milestones gave me a good plan to plan my progress around, and plan what I would try to accomplish that day. The permanent groups was also pretty helpful. Being able to keep up with certain people over time gave me a better understanding and appreciation of what they were doing, and gave me some insights that I eventually used in the construction of my program.
+
+There are a number of things that I would like to address in this program for the future. Overall, I think the program runs well (Cloud 9 does run into memory problems if the data set is too big, but I don’t think that is avoidable with large data sets), but it would be nice to give some flexibility to the number of data sets of the program- especially in cases where more data sets need to be analyzed as a whole. While I picked 5 as it was both a clean number and was enough points to see a clear trend (as well as the computation time), an actual data analyst may need to see upwards of 100 points to have a better picture of what is going on. I do have a vague idea of implementing this (perhaps through making the file-reading process into a function, and having the user expressing how many data files they want to use initially), but it will have to wait until after this project is over. Also, the histogram scaling is currently fixed- while I may be able to set it to an acceptable value for each one, the fact that it doesn’t automatically scale severely limits its scope. I would really like to implement a function or equation to calculate how the value should change, but due to limited time, testing values out over and over again would be too time-efficient for this version of the project.
```

Comment:
FYI the `csv` and other modules can get around the memory limits by not reading the entire file into memory at once.  We didn't go over anything about efficiency but pay attention to this in your future explorations.